src/main.js

```diff
@@ -0,0 +1,270 @@
+import {
+  Cartesian3,
+  Cartesian2,
+  Math as CesiumMath,
+  Terrain,
+  Viewer,
+  createOsmBuildingsAsync,
+  ImageryLayer,
+  IonWorldImageryStyle,
+  JulianDate,
+  HeadingPitchRoll,
+  Transforms,
+  GeoJsonDataSource,
+  HeadingPitchRange,
+  VerticalOrigin,
+  HorizontalOrigin,
+  HeightReference,
+  Color,
+  Ion,
+  defined,
+  ScreenSpaceEventType,
+  ScreenSpaceEventHandler,
+} from "cesium";
+import "cesium/Build/Cesium/Widgets/widgets.css";
+import "./style.css";
+
+// Step 1.2: Add you default Cesium ion access token
+Ion.defaultAccessToken = "your_ion_token_here";
+
+// Step 1.3: Initialize the Cesium Viewer in the HTML element with the
+// `cesiumContainer` ID and visualize terrain
+const viewer = new Viewer("cesiumContainer", {
+  terrain: Terrain.fromWorldTerrain(),
+  infoBox: false,
+});
+
+// Step 1.4: Add aerial imagery later with labels
+viewer.imageryLayers.add(
+  ImageryLayer.fromWorldImagery({
+    style: IonWorldImageryStyle.AERIAL_WITH_LABELS,
+  }),
+);
+
+// Step 1.5: Add Cesium OSM Buildings, a global 3D buildings layer.
+createOsmBuildingsAsync().then((buildingTileset) => {
+  viewer.scene.primitives.add(buildingTileset);
+});
+
+// Step 1.6: Enable lighting the globe, set time of day, and turn on animation sped up 60x
+viewer.scene.globe.enableLighting = true;
+const customTime = JulianDate.fromDate(
+  new Date(Date.UTC(2025, 5, 10, 3, 0, 0)),
+);
+viewer.clock.currentTime = customTime;
+viewer.clock.shouldAnimate = true;
+viewer.clock.multiplier = 60;
+
+// Step 1.7: Fly the camera to San Francisco at the given longitude, latitude, and height
+// and orient the camera at the given heading and pitch
+function setCamera() {
+  const position = Cartesian3.fromDegrees(-122.4075, 37.655, 400);
+  const heading = CesiumMath.toRadians(310.0);
+  const pitch = CesiumMath.toRadians(-10.0);
+  const range = 250;
+  const hpr = new HeadingPitchRange(heading, pitch, range);
+
+  viewer.camera.lookAt(position, hpr);
+}
+setCamera();
+
+// Step 2.1: Upload 3D model to the scene
+const position = Cartesian3.fromDegrees(-122.4875, 37.705, 300);
+
+function addModel(position) {
+  viewer.entities.removeAll();
+
+  const heading = CesiumMath.toRadians(135);
+  const pitch = 0;
+  const roll = 0;
+  const hpr = new HeadingPitchRoll(heading, pitch, roll);
+  const orientation = Transforms.headingPitchRollQuaternion(position, hpr);
+
+  viewer.entities.add({
+    name: "CesiumBalloon",
+    position: position,
+    orientation: orientation,
+    model: {
+      uri: "./src/CesiumBalloon.glb",
```

Comment:
Did we want to upload this model to ion as part of the workshop? If so, should the code here reflect that? Or do we prefer that it work out of the box?